backends/views.py

```diff
@@ -24,4 +26,13 @@ def complete(request, *args, **kwargs):
         request.session[key] = backend_state
 
     # Continue with social_core pipeline
-    return social_complete(request, *args, **kwargs)
+    social_complete_rtn = social_complete(request, *args, **kwargs)
+
+    # Update redis cache if user had invalid credentials
+    con = get_redis_connection("redis")
+    user_key = FIELD_USER_ID_BASE_STR.format(request.user.id)
+    if con.hexists(CACHE_KEY_FAILURE_NUMS_BY_USER, user_key):
+        con.hdel(CACHE_KEY_FAILURE_NUMS_BY_USER, user_key)
+        con.srem(CACHE_KEY_FAILED_USERS_NOT_TO_UPDATE, user_key)
+
+    return social_complete_rtn
```

Comment:
Can we call `social_complete(request, *args, **kwargs)` here? If there's no good reason to call it before the redis update vs after, we should just call it here so our code is grouped together

Reply:
the redis operations should be after we call `social_complete` because the user might be logged out at that point.